libvast/src/chunk.cpp

```diff
@@ -45,24 +56,20 @@ chunk_ptr chunk::mmap(const path& filename, size_t size, size_t offset) {
   ::close(fd);
   if (map == MAP_FAILED)
     return {};
-  auto deleter = [](char* buf, size_t n) { ::munmap(buf, n); };
-  return make(size, reinterpret_cast<char*>(map), deleter);
+  auto deleter = [=] { ::munmap(map, size); };
+  return make(size, reinterpret_cast<value_type*>(map), deleter);
 }
 
 chunk::~chunk() {
-  VAST_ASSERT(deleter_);
-  deleter_(data_, size_);
-}
-
-char* chunk::data() {
-  return data_;
+  if (deleter_)
```

Comment:
The comment in the header file explains the rationale: an absent deleter allows for creating shallow chunks. I currently use this for unit testing. However, I think it's obsolete because one can always perform a two-step by wrapping data into a `span<const byte>` and then creating a chunk from it. Then the chunk will only own the span. So yeah, we probably should ensure that a deleter is not default-constructed.